src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1814,13 +1839,16 @@ public static void shutdownAndWait(long timeout, TimeUnit units) throws Interrup
      * @param indexQueryPlan index query plan used in the read command
      * @param level consistency level of read command
      */
-    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints, Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel level)
+    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints, Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel level, boolean allowsFiltering)
```

Comment:
Nit: break the long line and add the new `allowsFiltering` param to the method's JavaDoc.